reference/test/solver/chebyshev_kernels.cpp

```diff
@@ -0,0 +1,472 @@
+// SPDX-FileCopyrightText: 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/solver/chebyshev_kernels.hpp"
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/solver/chebyshev.hpp>
+#include <ginkgo/core/solver/gmres.hpp>
+#include <ginkgo/core/stop/combined.hpp>
+#include <ginkgo/core/stop/iteration.hpp>
+#include <ginkgo/core/stop/residual_norm.hpp>
+
+#include "core/test/utils.hpp"
+
+template <typename T>
+class Chebyshev : public ::testing::Test {
+protected:
+    using value_type = T;
+    using Mtx = gko::matrix::Dense<value_type>;
+    using Solver = gko::solver::Chebyshev<value_type>;
+    Chebyshev()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(gko::initialize<Mtx>(
+              {{0.9, -1.0, 3.0}, {0.0, 1.0, 3.0}, {0.0, 0.0, 1.1}}, exec)),
+          // Eigenvalues of mtx are 0.9, 1.0 and 1.1
+          chebyshev_factory(
+              Solver::build()
+                  .with_criteria(
+                      gko::stop::Iteration::build().with_max_iters(30u),
+                      gko::stop::ResidualNorm<value_type>::build()
+                          .with_reduction_factor(r<value_type>::value))
+                  .with_foci(value_type{0.9}, value_type{1.1})
+                  .on(exec))
+    {}
+
+    std::shared_ptr<const gko::ReferenceExecutor> exec;
+    std::shared_ptr<Mtx> mtx;
+    std::unique_ptr<typename Solver::Factory> chebyshev_factory;
+};
+
+TYPED_TEST_SUITE(Chebyshev, gko::test::ValueTypes, TypenameNameGenerator);
+
+
+TYPED_TEST(Chebyshev, KernelInitUpdate)
+{
+    using value_type = typename TestFixture::value_type;
+    using Mtx = typename TestFixture::Mtx;
+    value_type alpha(0.5);
+    auto inner_sol = gko::initialize<Mtx>(
```

Comment:
those vectors seem to be repeated quite often, maybe pull them into the test class.